src/sentry/static/sentry/app/api.tsx

```diff
@@ -436,21 +436,31 @@ export class Client {
         let responseJSON: any;
         let responseText: any;
 
-        // Try to get JSON out of the response no matter the status
-        try {
-          responseJSON = await response.json();
-        } catch {
-          // No json came out.. too bad
-        }
+        const {status, statusText} = response;
+        let {ok} = response;
 
         // Try to get text out of the response no matter the status
         try {
-          responseText = await responseClone.text();
+          responseText = await response.text();
         } catch {
           // No text came out.. too bad
         }
 
-        const {ok, status, statusText} = response;
+        const responseContentType = response.headers.get('content-type');
+        const isResponseJSON = responseContentType?.includes('json');
+
+        const isStatus3XX = status >= 300 && status < 400;
+        if (status !== 204 && !isStatus3XX) {
```

Comment:
@Zylphrex If status if `204`, would `responseText` need to be `undefined`? It's usually an empty string in this case.

Reply:
I agree, I feel like that is outside the scope of this change which was intended to ensure invalid JSON responses are handled correctly. I'll look into a follow-up that addresses this.